### ISSUE TITLE
Keyboard issues

### ISSUE THREAD

**Issue by diegosanchezr** [badoo/Chatto#116]
Related: https://github.com/badoo/Chatto/pull/115

B)
1. Go to a chat conversation
2. Tap on textfield to open keyboard
3. Press return multiple times (until the textfield grows enough to show scroll indicators, and touches the navigation bar)
4. Reveal/Hide the predictive text bar above the keyboard
iOS 9: Cursor jumps slightly

E)
1. Go to a chat conversation
2. Tap on textfield to open keyboard
3. Paste very long text (so textfield grows enough to show scroll indicators, and touches the navigation bar)
3. Open Chinese (Traditional) - Zhuyin keyboard
4. Open full-screen chinese keyboard ( tap on ^ )
5. Drag the keyboard down by swiping down

Full-screen chinese keyboard will be misplaced. It will be in the middle of the screen ( looks like as much as the textfield grows once it's placed at the bottom )

F) 
1. Go to a chat conversation
2. Tap on textfield to open keyboard
3. Long press on a text message with a date
4. Choose "Create event" from action sheet
5. In the create event screen dismiss the keyboard
6. Cancel event creation

Input field is placed in the middle of the screen

**Comment by zwang** [comment]
For B, can we limit the maximum height for the input textfield? If the user need to enter that much text, then it is just a matter time that user need to scroll in the input textfield anyway.

**Comment by diegosanchezr** [comment]
Yes, it would be nice if we can configure with some percentage (0% grows until nav bar, 50% ~ middle of the screen) or some number of lines depending on total height as well.

**Comment by nilsgrabenhorst** [comment]
For F:
The keyboard notifications of the EKEventEditViewController are sent before viewWillDisappear is called on BaseChatViewController. The KeyboardTracker is still tracking, therefore the input field moves up. However, tracking stops when the BaseChatViewController disappears, so the tracker now ignores the UIKeyboardWillHide notification sent before the EKEventEditViewController disappears.

I’m not sure how to deal with this issue. If tracking is never switched off it kind of half-works: The input field has been moved back into place before the event edit view controller disappears. But the layout change of the input field is still visible before the edit view controller animates in.

Maybe tracking could be switched off when the textView delegate’s textView(_:shouldInteractWith:in:interaction:) is called?
